dll/win32/browseui/parsecmdline.cpp

```diff
@@ -382,8 +382,43 @@ SHExplorerParseCmdLine(ExplorerCommandLineParseResults * pInfo)
             {
                 // Or just a plain old string.
 
+                WCHAR szPath[MAX_PATH];
+
+                if (!wcsncmp(strField, L"\\", 1))
+                {
+                    // this is root path
+                    if (GetCurrentDirectory(_countof(szPath), szPath))
+                    {
+                        if (!wcsncmp(szPath + 1, L":", 1))
+                        {
+                            // this is local path
+                            szPath[2] = 0;
+                            StringCchCatW(szPath, _countof(szPath), strField);
+                            StringCchCopyW(strField, _countof(strField), szPath);
+                        }
+                        else if (!wcsncmp(szPath, L"\\\\", 2))
+                        {
+                            // this is network path
+                            FIXME("Network path detected: %S", szPath);
+                        }
+                    }
+                }
+
                 LPITEMIDLIST pidlPath = ILCreateFromPathW(strField);
 
+                if (!pidlPath &&
```

Comment:
It seems like a simple GetFullPathName call before calling ILCreateFromPath might be able to simplify all this.